docs/quarantine.md

```diff
@@ -38,7 +38,7 @@ In the following failure rates are calculating taking into account periodics and
 presubmits that are executed on merged code (on tide merge batches).
 
 A test must be put in quarantine when any of these conditions is met:
-* It has a failure rate higher than 5% in the last two weeks.
+* It has a failure rate higher than 0.5% in the last two weeks.
```

Comment:
This may be too low to measure: it means one failure for 200 executions. When would one obtain such statistics?

I would not mind to keep a test that failed twice over two weeks if nothing else is failing. Can we ask to quarantine the most frequent fails, until the pass rate goes above 95% ?

Reply:
> I would not mind to keep a test that failed twice over two weeks if nothing else is failing. Can we ask to quarantine the most frequent fails, until the pass rate goes above 95% ?

@dankenigsberg I think this is absolutely a great goal, but in my opinion optimizing for a 95% total test suite pass rate (rather than 95% test case pass rate) right now is too aggressive. To hit a 95% total test suite pass rate today, we might have to disable so many tests that we begin to expose the project to further instability. It's the whole "we're working on an airplane in flight" analogy, where if we are too aggressive we risk falling out of the sky. 

I'd like to see 95% total test suite pass rate as our goal, and our quarantine policy to stick with the previously mentioned "quarantine a test case after x% of failures over y time". We will start to see improvements. For example, this threshold would definitely have caught the issues that caused the merge issues two weeks ago. There we had test cases failing more like 10-30% of the time.

Over time, we should begin tweaking this per test case percentage with the goal of getting it down to 1% and even 0.5% per test case failure rates. I believe that will eventually hit your 95% total test suite pass rate, but in an incremental way that introduces less risk.

thoughts?

Reply:
From my POV it would be better to set rates as goals instead of concrete numbers without references to how many executions were done. I also think that this is an initial policy that should not be set on stone, we should adapt it to our needs over time, we can modify the concrete numbers as needed.

I agree that 0.5% is too low to measure (with the current execution rate, just one failure would represent a much higher failure rate), I've reverted it to 5%, let me know if that is ok.

Reply:
@davidvossel I'm all for gradual improvements, not pies-in-the-skies. I'd like us all to never accept again a condition where doc PRs fail more than 5% of the times.

for example, this very PR has
```
Test name | Commit | Details | Rerun command
-- | -- | -- | --
pull-kubevirt-e2e-k8s-1.19 | e57c442 | link | /test pull-kubevirt-e2e-k8s-1.19
pull-kubevirt-e2e-k8s-1.18 | e57c442 | link | /test pull-kubevirt-e2e-k8s-1.18
pull-kubevirt-unit-test | e5a5c4f | link | /test pull-kubevirt-unit-test
```
It makes no sense at all. Our procedures must ensure that people are working to fix this. To me, this quarantine document is a mechanism to pull in many people to work on this critical point in our project. How can we ensure that people focus on CI improvement until the failure rate falls below 5%?

Reply:
I've added an additional section to the doc with the goal of the methodology set to whole test suite failure rate at 10% (for considering the suite as stable) and per test failure rate to 5% (for the concrete test to be proposed for quarantine), let me know wdyt.

Reply:
@davidvossel, I understand. But I would like this document to express that having 17 individual tests each failing 4% of the time we get to 96%**17=50% fail rate, **requires** an aggressive stance.

Reply:
@dankenigsberg I've added a note about how the whole suite passing rate depends on the individual tests passing rate, PTAL.